addons/FlashCards/src/presenter.js

```diff
@@ -666,7 +698,11 @@ function AddonFlashCards_create(){
         presenter.state = parsedState.state;
 
         presenter.updateVisibility();
-        presenter.showCard(presenter.state.currentCard);
+        if (!presenter.configuration.randomizeOrder) {
+            presenter.showCard(presenter.state.currentCard);
+        }   else {
+            presenter.state.currentCard = 1;

```

Comment:
Wybacz. Moja analiza była niewystarczająca dla setState. Konieczna jest jeszcze jedna poprawka.
Po uruchomieniu komendy ShowOnlyFavourites mają pokazywać się tylko ulubione karty. Jeżeli na sztywno tutaj wybierzesz pierwszą kartę to nie będzie to zawsze ta pierwsza ulubiona. Skutkuje to pokazaniem nieulubionej karty oraz błąd w indeksacji dla następnej karty czyli:
![image](https://github.com/icplayer/icplayer/assets/79645404/c54d1276-7210-4816-a3e8-0210b608822a)